packages/atomic/src/utils/accessibility-utils.tsx

```diff
@@ -96,3 +96,51 @@ export function FocusTarget() {
     };
   };
 }
+
+function isFocusable(element: Element) {
+  // Source: https://stackoverflow.com/a/30753870
+  if (element.getAttribute('tabindex') === '-1') {
+    return false;
+  }
+  if (element.hasAttribute('tabindex')) {
+    return true;
+  }
+  if (element.getAttribute('contentEditable') === 'true') {
+    return true;
+  }
+  switch (element.tagName) {
+    case 'A':
+    case 'AREA':
+      return element.hasAttribute('href');
+    case 'INPUT':
+    case 'SELECT':
+    case 'TEXTAREA':
+    case 'BUTTON':
+      return !element.hasAttribute('disabled');
+    case 'IFRAME':
+      return true;
+    default:
+      return false;
+  }
+}
+
+export function getFirstFocusableDescendant(
+  element: Element
+): HTMLElement | null {
+  if (isFocusable(element)) {
```

Comment:
I'm using a recursive function instead of `querySelector` in order to traverse the shadowDOM and slots.